src/defaultCredentials.ts

```diff
@@ -0,0 +1,89 @@
+import * as fs from "fs";
+import * as path from "path";
+
+import * as api from "./api";
+import { configstore } from "./configstore";
+import * as logger from "./logger";
+
+interface RefreshTokenCredential {
+  client_id: string;
+  client_secret: string;
+  refresh_token: string;
+  type: string;
+}
+
+/**
+ * Creates Cloud application default credentials using the authorization from
+ * 'firebase login'.
+ */
+export class DefaultCredentials {
+  /**
+   * Get a path to the application default credentials JSON file.
+   */
+  public async getPathAsync(): Promise<string | undefined> {
```

Comment:
Is it worth writing any unit tests for exported methods in this file?

Reply:
Done, good idea.